### PR TITLE
Containerize application with docker

### PR DESCRIPTION
**Pull request by sangvhh**
#### Issue

-   Reduce size of image by using multistage dockerfile.

#### Changes

-   [ ] Fix bug
-   [x] New feature
-   [ ] Documentation
-   [ ] Backward incompatible change

#### Testing

-   Manual test.

#### Checklist

-   [x] Checked README.md.
-   [ ] Checked CHANGELOG.md
-   [ ] Checked comments.

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -9,14 +7,21 @@ COPY go.sum .
...
-COPY cmd cmd/
-COPY internal internal/
-COPY pkg pkg/
-COPY web web/
+COPY . ./
 
-RUN go build -o /server ./cmd/server/*.go
-RUN go build -o /database ./cmd/database/*.go
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -o /server ./cmd/server/*.go \
+ && CGO_ENABLED=0 go build -ldflags="-w -s" -o /database ./cmd/database/*.go
```
Comment (huykingsofm) [inline build/Dockerfile:13]: What purpose of CGO_ENABLED=0 and -ldflag?

File: build/Dockerfile [hunk 2]
```diff
@@ -9,14 +7,21 @@ COPY go.sum .
...
+
+COPY --from=build /server /database /
+COPY --from=build /xyauth/configs /configs
+COPY --from=build /xyauth/web /web
 
-ENTRYPOINT [ "/server" ]
 EXPOSE 8443
+
+ENTRYPOINT [ "/server" ]
+
```
Comment (huykingsofm) [inline build/Dockerfile]: Remove a break line.

**Review by huykingsofm — CHANGES_REQUESTED** [review]
Following the guide in [how to remove intermediate images](https://stackoverflow.com/questions/50126741/how-to-remove-intermediate-images-from-a-build-after-the-build) to modify Makefile.

File: build/Dockerfile (hunk 1 above)
Comment (sangvhh) [reply build/Dockerfile:13]: CGO_ENABLED=0 means that the resulting binary will not be linked to any C libraries.
You will get the smallest binaries if you compile with -ldflags '-w -s'.
-s option disable symbol table and  -w option disable DWARF generation.

**Review by huykingsofm — APPROVED** [review]

**Pull request merged**